src/Psalm/Internal/Provider/StatementsProvider.php

```diff
@@ -155,10 +155,6 @@ public function getStatementsForFile(
 
             $existing_statements = $this->parser_cache_provider->loadExistingStatementsFromCache($file_path);
 
-            if ($existing_statements && !$existing_statements[0] instanceof PhpParser\Node\Stmt) {
```

Comment:
I made a check that went a little too far so I had to revert it, but it saw this as redundant. I couldn't find the reason for this, it seems redundant indeed